api/provider/aws/load_balancer_create.go

```diff
@@ -90,41 +95,116 @@ func (l *LoadBalancerProvider) Create(req models.CreateLoadBalancerRequest) (str
+		if err != nil {
+			return "", err
+		}
 
-	if err := l.createLoadBalancer(
-		fqLoadBalancerID,
-		scheme,
-		securityGroupIDs,
-		subnets,
-		listeners); err != nil {
-		return "", err
+		if err := l.createLoadBalancer(
+			fqLoadBalancerID,
+			scheme,
+			securityGroupIDs,
+			subnets,
+			listeners); err != nil {
+			return "", err
+		}
+
+		if req.HealthCheck == (models.HealthCheck{}) {
+			req.HealthCheck = config.DefaultLoadBalancerHealthCheck()
+		}
+
+		if err := l.updateCLBHealthCheck(fqLoadBalancerID, req.HealthCheck); err != nil {
+			return "", err
+		}
 	}
 
-	if req.HealthCheck == (models.HealthCheck{}) {
-		req.HealthCheck = config.DefaultLoadBalancerHealthCheck()
+	if req.LoadBalancerType.Equals(models.ApplicationLoadBalancerType) {
+		lb, err := l.createApplicationLoadBalancer(
+			fqLoadBalancerID,
+			scheme,
+			securityGroupIDs,
+			subnets)
+		if err != nil {
+			return "", err
+		}
+
+		if req.HealthCheck == (models.HealthCheck{}) {
+			req.HealthCheck = config.DefaultLoadBalancerHealthCheck()
+		}
+
+		targetGroupName := fqLoadBalancerID
+		tg, err := l.createTargetGroup(targetGroupName, req.HealthCheck)
+		if err != nil {
+			return "", err
+		}
+
+		if err := l.createListener(lb.LoadBalancerArn, tg.TargetGroupArn, req.Ports); err != nil {
+			return "", err
+		}
 	}
 
-	healthCheck := &elb.HealthCheck{
-		Target:             aws.String(req.HealthCheck.Target),
-		Interval:           aws.Int64(int64(req.HealthCheck.Interval)),
-		Timeout:            aws.Int64(int64(req.HealthCheck.Timeout)),
-		HealthyThreshold:   aws.Int64(int64(req.HealthCheck.HealthyThreshold)),
-		UnhealthyThreshold: aws.Int64(int64(req.HealthCheck.UnhealthyThreshold)),
+	return loadBalancerID, nil
+}
+
+func (l *LoadBalancerProvider) createTargetGroup(groupName string, healthCheck models.HealthCheck) (*alb.TargetGroup, error) {
+	input := &alb.CreateTargetGroupInput{}
+	input.SetName(groupName)
+	input.SetPort(config.DefaultTargetGroupPort)
+	input.SetProtocol(config.DefaultTargetGroupProtocol)
+	input.SetVpcId(l.Config.VPC())
+	input.SetTargetType(alb.TargetTypeEnumIp)
+
+	// set health check
+	input.SetHealthCheckPath(healthCheck.Path)
+	input.SetHealthCheckIntervalSeconds(int64(healthCheck.Interval))
+	input.SetHealthCheckTimeoutSeconds(int64(healthCheck.Timeout))
+	input.SetHealthyThresholdCount(int64(healthCheck.HealthyThreshold))
+	input.SetUnhealthyThresholdCount(int64(healthCheck.UnhealthyThreshold))
+
+	if err := input.Validate(); err != nil {
+		return nil, err
 	}
 
-	if err := l.updateHealthCheck(fqLoadBalancerID, healthCheck); err != nil {
-		return "", err
+	output, err := l.AWS.ALB.CreateTargetGroup(input)
+	if err != nil {
+		return nil, err
 	}
 
-	if err := l.createTags(loadBalancerID, req.LoadBalancerName, req.EnvironmentID); err != nil {
-		return "", err
+	return output.TargetGroups[0], nil
+}
+
+func (l *LoadBalancerProvider) createListener(loadBalancerArn, targetGroupArn *string, ports []models.Port) error {
```

Comment:
abbreviations are all uppers in go: `Arn -> ARN`

Reply:
✅ 

as a side note - I got a warning about this from the IDE but hesitantly decided to stay consistent with what is aws go sdk provides (Arn)